### PR TITLE
Plugin external metadata_edns0

### PR DESCRIPTION
**Pull request by rajansandeep**
Initial commit
Needs CI tests.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+~~~
+metadata_edns0 {
+      client_id 0xffed address
+      group_id 0xffee hex 16 0 16
+      <label> <id> <encoded-format> <params of format ...>
+}
+~~~
+
+So far, only 'hex' format is supported with params <length> <start> <end>
```
Comment (fturib) [inline README.md]: That comment seems wrong to me. you have three format supported. However only `hex` has extra parameters

File: metadata_edns0.go [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+func (m metadataEdns0) Name() string { return "metadata_edns0" }
+
+func (m *metadataEdns0) Metadata(ctx context.Context, state request.Request) context.Context {
+	return m.registerFuncExtractEDNS0(ctx, state.Req)
+}
+
+func (m *metadataEdns0) registerFuncExtractEDNS0(ctx context.Context, r *dns.Msg) context.Context {
+	o := r.IsEdns0()
+	if o == nil {
+		return nil
```
Comment (fturib) [inline metadata_edns0.go]: I think you should not return nil here, but `ctx`. See plugin metadata, function `ServeDNS`

**Review by fturib — CHANGES_REQUESTED** [review]
Some few changes before we can merge this first version.
Then we will need to add the CI and some tests running, including a test with a real execution of ServeDNS with ctx and data in the query.

**Comment by rajansandeep** [comment]
@fturib I addressed your comments.
I'll merge this now.

**Pull request merged**